sources/packages/api/src/services/application/application.service.ts

```diff
@@ -544,21 +608,39 @@ export class ApplicationService extends RecordDataModelService<Application> {
       .where("application.student_id = :studentId", { studentId })
       .andWhere("application.applicationStatus != :overwrittenStatus", {
         overwrittenStatus: ApplicationStatus.overwritten,
-      })
-      .orderBy(
+      });
+
+    // sorting
+    if (
+      sortField &&
+      sortField !== "status" &&
+      databaseFieldOfApplicationDataTable(sortField)
+    ) {
+      application.orderBy(
+        databaseFieldOfApplicationDataTable(sortField),
+        sortOrder,
+      );
+    } else {
+      application.orderBy(
         `CASE application.applicationStatus
-            WHEN '${ApplicationStatus.draft}' THEN 1
-            WHEN '${ApplicationStatus.submitted}' THEN 2
-            WHEN '${ApplicationStatus.inProgress}' THEN 3
-            WHEN '${ApplicationStatus.assessment}' THEN 4
-            WHEN '${ApplicationStatus.enrollment}' THEN 5
-            WHEN '${ApplicationStatus.completed}' THEN 6
-            WHEN '${ApplicationStatus.cancelled}' THEN 7
-            ELSE 8
-          END`,
-      )
-      .addOrderBy("application.applicationNumber")
-      .getMany();
+              WHEN '${ApplicationStatus.draft}' THEN 1
+              WHEN '${ApplicationStatus.submitted}' THEN 2
+              WHEN '${ApplicationStatus.inProgress}' THEN 3
+              WHEN '${ApplicationStatus.assessment}' THEN 4
+              WHEN '${ApplicationStatus.enrollment}' THEN 5
+              WHEN '${ApplicationStatus.completed}' THEN 6
+              WHEN '${ApplicationStatus.cancelled}' THEN 7
+              ELSE 8
+            END`,
+        sortOrder,
+      );
+    }
+
+    // pagination
+    application.limit(pageLimit).offset(page * pageLimit);
```

Comment:
I believe we discussed this, To use take/skip when using getManyAndCount, and limit/offset when fetching getRawManyAndCount. Please tell, if you find any issues during loading?
https://stackoverflow.com/questions/68468192/difference-between-limit-and-take-in-typeorm

Reply:
yup, take and skip is causing issue, 
![image](https://user-images.githubusercontent.com/77353155/146428723-d0c0a7dd-a138-45d4-b9dd-01f5ecc9e353.png)